### PR TITLE
Start using daohaus metadata when populating DAO information

### LINKED ISSUE #646: [MyMeta] Metadata from daohaus.club for DAO images and names
Some DAOs listed in the HausDAO subgraph do not have titles. This is a "bug" (as highlighted in this comment: https://github.com/HausDAO/daohaus-supergraph/issues/44#issuecomment-858844186). In reality, new DAOs spun up on a Moloch contract do not have titles. The recommendation is to move to the DaoHaus Metadata API.

Using the metadata API, the following fields would be immediately useful:

* `name`
* `avatarImage`

Potentially useful in the future:

* `description`
* `links.website`

Here is the output for MetaCartel metadata (https://data.daohaus.club/dao/0xb152b115c94275b54a3f0b08c1aa1d21f32a659a):

```json
[
    {
        "contractAddress": "0xb152b115c94275b54a3f0b08c1aa1d21f32a659a",
        "network": "xdai",
        "name": "MetaCartel",
        "description": "If you want to go fast, go alone. If you want to go far, go together.",
        "purpose": "Grants",
        "version": "2.1",
        "slug": "metacartel-v2-hd",
        "tags": [
            "Ethereum",
            "Clubs"
        ],
        "links": {
            "website": "https://metacartel.org",
            "twitter": "Meta_Cartel",
            "discord": "https://discord.gg/vdeUzHwz2R",
            "telegram": "",
            "medium": "",
            "other": ""
        },
        "avatarImg": "QmUYPn3g6gAxRVAYjhFKzByGn4B5yvVTX12VENvz6bgwiH",
        "customTheme": {
            "primary500": "#9670d1",
            "primaryAlpha": "rgba(150,112,209,0.9)",
            "secondary500": "#6c1096",
            "secondaryAlpha": "rgba(108,16,150,0.75)",
            "bg500": "#858585",
            "bgAlpha": "#03061B",
            "bgOverlayOpacity": 0.75,
            "headingFont": "Rubik",
            "bodyFont": "Merriweather",
            "monoFont": "JetBrains Mono",
            "avatarImg": "/static/media/Daohaus__Castle--Dark.3084fa93.svg",
            "bgImg": "QmQ9fHmZMR2SRyFxuABYPNxCGpyN839kZ1hVcEzPwF4F9E"
        },
        "daosquarecco": 0,
        "customTerms": {
            "proposals": "Proposals",
            "proposal": "Proposal",
            "bank": "Bank",
            "members": "Members",
            "member": "Member",
            "boosts": "Apps",
            "boost": "App",
            "settings": "Settings",
            "ragequit": "Rage Quit",
            "guildKick": "Guild Kick",
            "minion": "Minion",
            "minions": "Minions"
        },
        "boosts": {
            "customTheme": {
                "active": true,
                "metadata": {
                    "primary500": "#9670d1",
                    "primaryAlpha": "rgba(150,112,209,0.9)",
                    "secondary500": "#6c1096",
                    "secondaryAlpha": "rgba(108,16,150,0.75)",
                    "bg500": "#858585",
                    "bgAlpha": "#03061B",
                    "bgOverlayOpacity": 0.75,
                    "headingFont": "Rubik",
                    "bodyFont": "Merriweather",
                    "monoFont": "JetBrains Mono",
                    "avatarImg": "/static/media/Daohaus__Castle--Dark.3084fa93.svg",
                    "bgImg": "QmQ9fHmZMR2SRyFxuABYPNxCGpyN839kZ1hVcEzPwF4F9E",
                    "daoMeta": {
                        "proposals": "Proposals",
                        "proposal": "Proposal",
                        "bank": "Bank",
                        "members": "Members",
                        "member": "Member",
                        "boosts": "Apps",
                        "boost": "App",
                        "settings": "Settings",
                        "ragequit": "Rage Quit",
                        "guildKick": "Guild Kick",
                        "minion": "Minion",
                        "minions": "Minions"
                    }
                }
            },
            "notificationsLevel1": {
                "active": true,
                "metadata": [
                    {
                        "type": "discord",
                        "channelId": "769698426837139466",
                        "active": true,
                        "actions": [
                            "votingPeriod",
                            "rageQuit",
                            "newProposal"
                        ]
                    }
                ]
            },
            "snapshot": {
                "active": true,
                "metadata": {
                    "space": "metacartel.eth"
                }
            },
            "mintGate": {
                "active": true,
                "metadata": {}
            }
        },
        "allies": [
            {
                "allyType": "uberHausBurner",
                "isParent": false,
                "allyNetwork": "mainnet",
                "ally": "0xee629a192374caf2a72cf1695c485c5c89611ef2"
            },
            {
                "allyType": "uberHausBurner",
                "isParent": false,
                "allyNetwork": "mainnet",
                "ally": "0xee629a192374caf2a72cf1695c485c5c89611ef2"
            }
        ]
    }
]
```

---

References:

* Issue on subgraph repo: https://github.com/HausDAO/daohaus-supergraph/issues/44#issuecomment-858844186
* Discussion on DAOhaus Discord: https://discord.com/channels/709210493549674598/735524730328711188/852600235468259388

Related issues: #549, https://github.com/MetaFam/TheGame/pull/643

### PR DESCRIPTION
**Pull request by alalonde**
## Overview

Changes the daohaus lookup code to fetch DAO metadata from an undocumented daohaus "API" 

Closes #646, #838

## Follow up Improvement Ideas

- Cache this data appropriately.  This makes the DAO lookup even slower since we're doing an additional HTTP request for each individual DAO. (See #997)
- Implement backend and UI for defining a whitelist of DAOs you want to include in your profile

## Assets

daohaus

### CODE REVIEW

**Comment by alalonde** [comment]
vercel is failing due to the backend schema changing.  Not sure why the GCD action didn't deploy...

**Comment by dan13ram** [comment]
This looks so cool. May not be necessary in this PR, but we should add a tag to each DAO to display which network it is on?
The same for NFTs too would be cool (once we start displaying NFTs from other networks as well)

**Review by dan13ram — APPROVED** [review]
Great work! 

I see that the svg chain icons aren't being used. I think we could start using them instead of the pngs. Otherwise everything else is 👌🏽

File: packages/web/utils/daoHelpers.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+  if (chain?.toLowerCase().includes('xdai')) return xDaiImage;
+  if (chain?.toLowerCase().includes('polygon')) return polygonImage;
+  return ethereumImage;
+};
+
+export const getDaoLink = (chain?: string, address?: string): string | null => {
+  if (address && chain) {
+    const hexChainId = getHexChainId(chain);
+    address = address.toLowerCase(); // eslint-disable-line no-param-reassign
+    return `https://app.daohaus.club/dao/${hexChainId}/${address}`;
```
[the remark this answers is not in the document]
Comment (dan13ram) [reply packages/web/utils/daoHelpers.ts:17]: actually the method can be shortened to 
```
export const getDaoLink = (chain?: string, address?: string): string | null =>
  address && chain
    ? `https://app.daohaus.club/dao/${getHexChainId(chain)}/${address.toLowerCase()}`
    : null;
```

File: packages/design-system/src/icons/ChainIcon.tsx [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import { EthereumIcon } from './EthereumIcon';
+import { PolygonIcon } from './PolygonIcon';
+import { XDaiIcon } from './XDaiIcon';
+
+type Props = {
+  chain: string | undefined;
+};
+
+export const ChainIcon: React.FC<Props & IconProps> = ({ chain, ...props }) => {
```
[the remark this answers is not in the document]
Comment (alalonde) [reply packages/design-system/src/icons/ChainIcon.tsx:12]: Ha, yes, I added this to be able to show a little chain icon next to the DAO as well, but the various svgs rendered weird so I bailed on the idea.

**Comment by alalonde** [comment]
Alright, just made another rather large update. I removed the old chain images in favor of the new SVG one, and tweaked the SVGs so that they render better. I also tested and updated the /player/setup/memberships component to match the one for viewing players

File: packages/web/components/Player/PlayerGuild.tsx [hunk 3]
```diff
@@ -1,78 +1,56 @@
...
+type InternalGuildLinkProps = {
+  guildName: string;
+};
+
+export const InternalGuildLink: React.FC<InternalGuildLinkProps> = ({
+  guildName,
+  children,
+}) => (
+  <Link
+    role="group"
```
Comment (dan13ram) [inline packages/web/components/Player/PlayerGuild.tsx:32]: Why is there a role=group here ? I think its creating some issues.

File: packages/web/components/Player/PlayerGuild.tsx [hunk 4]
```diff
@@ -1,78 +1,56 @@
...
+);
+
+type DaoHausLinkProps = {
+  daoUrl: string | null;
+};
+
+export const DaoHausLink: React.FC<DaoHausLinkProps> = ({ daoUrl, children }) =>
+  daoUrl != null ? (
+    <Link
+      role="group"
```
Comment (dan13ram) [inline packages/web/components/Player/PlayerGuild.tsx:47]: Here as well.

File: packages/web/components/Player/PlayerGuild.tsx (hunk 3 above)
Comment (dan13ram) [reply packages/web/components/Player/PlayerGuild.tsx:32]: CPT2202191414-472x522

Comment (alalonde) [reply packages/web/components/Player/PlayerGuild.tsx:32]: Well, all I did was split these components into three discrete components without changing their functionality. So `role="group"` has been there all along.

Comment (dan13ram) [reply packages/web/components/Player/PlayerGuild.tsx:32]: oh right. still i think it shouldn't be there

Comment (alalonde) [reply packages/web/components/Player/PlayerGuild.tsx:32]: _groupHover is used in PlayerGuildMemberships as one of the `children`

Comment (dan13ram) [reply packages/web/components/Player/PlayerGuild.tsx:32]: Are you seeing those UI bugs on hover or its misbehaving only on my side?

Comment (alalonde) [reply packages/web/components/Player/PlayerGuild.tsx:32]: I see it on Chrome but not on FF

**Pull request merged**